File: 13xx/1381.py
class CustomStack(object):
    def __init__(self, maxSize):
        self.size = maxSize
        self.end = 0
        self.stack = [None for i in range(maxSize)]
        

    def push(self, x):
        if self.end == self.size:
            return
        self.stack[self.end] = x
        self.end += 1


    def pop(self):
        if self.end == 0:
            return -1
        self.end -= 1
        return self.stack[self.end]


    def increment(self, k, val):
        for i in range(min(k, self.end)):
            self.stack[i] += val

File: 13xx/1381/1381.cs
public class CustomStack 
{
    private int[] stack;
    
    private int currentPos = -1;
    
    private int maxSize;

    public CustomStack(int maxSize) 
    {
        stack = new int[maxSize];
        this.maxSize = maxSize;
    }
    
    public void Push(int x) 
    {
        if (currentPos < maxSize - 1) 
            stack[++currentPos] = x;
    }
    
    public int Pop() 
    {
        if (currentPos < 0) 
            return -1;
        
        return stack[currentPos--];
    }
    
    public void Increment(int k, int val) 
    {
        for (int i = 0; i < k && i < maxSize; i++) 
            stack[i] += val;
    }
}